### PR TITLE
Collapse functionality

### PR DESCRIPTION
**Pull request by emreliman**

### CODE REVIEW

File: src/components/GridTable.vue [hunk 1]
```diff
@@ -20,59 +21,135 @@
...
+  },
 
-@Component
-export default class GridTable extends Vue {
-  @Prop() private table!: IRow;
+  setup(props,context){
+    console.log("context",context);
+    console.log("props.table", props.table)
+    let collapseTable= ref<IRow>([])
+    const watch_val = ref(false);
```
Comment (ismailarilik) [inline src/components/GridTable.vue]: In JS, general convention is to use camelCase over snake_case.

File: src/components/GridTable.vue [hunk 2]
```diff
@@ -20,59 +21,135 @@
...
+      if(!collapseTable.value)
+      return
+      
+      if(collapseTable.value.length === 0){
+        collapseTable.value = [...props.table as Array<Row>]
+        console.log("collapseTable",collapseTable.value)
+      }
+      
+    })
+    // onMounted(()=>{
```
Comment (ismailarilik) [inline src/components/GridTable.vue]: Those comments are not necessary.

File: src/components/GridTable.vue [hunk 3]
```diff
@@ -20,59 +21,135 @@
...
+    //   console.log("props.table 2", table)
+    //   collapseTable.value = table
+    //   console.log("collapseTable",collapseTable.value)
+    // })
+   
+  const button_id = (rowIndex:number)=>{
```
Comment (ismailarilik) [inline src/components/GridTable.vue]: Another snake_case variable name...

File: src/components/GridTable.vue [hunk 4]
```diff
@@ -20,59 +21,135 @@
...
 
-  // alignment returns the alignment for the cell at the given row and column index and the given alignment type
-  alignment(row?: Row, colIndex?: string, type?: string) {
+  const alignment = (row?: Row, colIndex?: string, type?: string) =>{
     return type === "vertical" ? ("baseline" as const) : ("left" as const);
   }
 
-  filteredCells(cells?: Array<any>) {
-    return cells?.filter(_ => true);
+  const show_button = (row: Row, colIndex: number, rowIndex: number) =>{
```
Comment (ismailarilik) [inline src/components/GridTable.vue]: And another one... Could you refactor all snake_case names as camelCase names?

File: tests/unit/GridTable.spec.ts [hunk 5]
```diff
@@ -20,7 +25,7 @@ describe('GridTable.vue', () => {
       stach.factset.protobuf.stach.v2.RowOrganizedPackage.create(db.data as unknown as stach.factset.protobuf.stach.v2.RowOrganizedPackage);
     table.value = pkg.tables.main.data?.rows;
 
-    wrapper = shallowMount(GridTable, {
+    wrapper = shallowMount(GridTable as any, {
```
Comment (ismailarilik) [inline tests/unit/GridTable.spec.ts:28]: Isn't there any solution for that?

File: tests/unit/GridTableCollapse.spec.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    let wrapper: Wrapper<Vue>;
+
+    beforeEach(() => {
+        const pkg = stach.factset.protobuf.stach.v2.RowOrganizedPackage.create(
+            db.data as unknown as stach.factset.protobuf.stach.v2.RowOrganizedPackage
+        );
+        table.value = pkg.tables.main.data?.rows;
+
+        //add isOpen every element of table
+        table.value?.forEach((item: ExtraIRow) => {
```
Comment (ismailarilik) [inline tests/unit/GridTableCollapse.spec.ts]: Duplicated logic; see below...

File: tests/unit/GridTableCollapse.spec.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            item["isOpen"] = false;
+        });
+        //add childlength to each row according to grouplevel property
+        table.value?.forEach((item: ExtraIRow, index: number) => {
```
Comment (ismailarilik) [inline tests/unit/GridTableCollapse.spec.ts:25]: Duplicated logic. They exist in App.vue, too. They might be moved to a function.

**Review by ismailarilik — COMMENTED** [review]
Great job! I just reviewed and added some comments to be considered to improve the general code quality. But if you don't have time, you can merge (You will probably have a technical debt though so it is a good practice to note it somewhere.).

File: tests/unit/GridTable.spec.ts (hunk 5 above)
Comment (emreliman) [reply tests/unit/GridTable.spec.ts:28]: I don't find any solution for this.

**Review by ismailarilik — APPROVED** [review]

**Pull request merged**